app/models/journaled/delivery.rb

```diff
@@ -56,6 +60,14 @@ def legacy_credentials_hash
     end
   end
 
+  def iam_assume_role_credentials
+    @iam_assume_role_credentials ||= Aws::AssumeRoleCredentials.new(
+      client: Aws::STS::Client.new(legacy_credentials_hash),
```

Comment:
This reads confusingly, but it's probably the right behavior - am I right that it means that legacy_credentials_hash will be empty if there are no legacy credentials present and therefore default to looking up creds from the env however it wants? This is sort of not-your-code kind of territory, but I wonder if we should rename the method to `legacy_credentials_if_present` or something so it's clear that it falls thru to nothing.

Reply:
yes, that's correct! and that seems like a fine alternative. I could also initialize the client in a separate method to make it even more clear it's initialized with empty creds?

```
def aws_sts_client
  if legacy_credentials_hash.any?
    Aws::STS::Client.new(legacy_credentials_hash)
  else
    Aws::STS::Client.new
  end
end
```

Reply:
that still relies on the legacy_credentials_hash having novel null-object-patterny behavior for a method name that doesn't reveal that (returning an empty enumerable instead of `nil` or erroring on lack of legacy creds) - I think we should either go to make legacy_credentials_hash return nil, error if nil and have a different huh method, or rename it? Your call for what feels tidiest in context.